From c95c3c0c24becb32e3743ce9690f390f1bd0604d Mon Sep 17 00:00:00 2001
From: Claudiu Manoil <claudiu.manoil@nxp.com>
Date: Mon, 21 Jan 2019 11:28:10 +0200
Subject: [PATCH 628/706] enetc: Use hweightX instead of __sw_hweightX for
 cross arch portability

GCC_VERSION=8.1.0 make.cross ARCH=ia64

All errors (new ones prefixed by >>):

   ERROR: "ia64_delay_loop" [drivers/spi/spi-thunderx.ko] undefined!
   ERROR: "numa_slit" [drivers/nvme/host/nvme-core.ko] undefined!
   ERROR: "__sw_hweight8" [drivers/net/wireless/mediatek/mt76/mt76.ko]
undefined!
   ERROR: "ia64_delay_loop" [drivers/net/phy/mdio-cavium.ko] undefined!
>> ERROR: "__sw_hweight8"
>> [drivers/net/ethernet/freescale/enetc/fsl-enetc.ko] undefined!
>> ERROR: "__sw_hweight64"
>> [drivers/net/ethernet/freescale/enetc/fsl-enetc.ko] undefined!

Signed-off-by: Claudiu Manoil <claudiu.manoil@nxp.com>
(cherry picked from commit d7e216c7e3f469a2d41d356a6443aeec5469740e)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/freescale/enetc/enetc_pf.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/freescale/enetc/enetc_pf.c b/drivers/net/ethernet/freescale/enetc/enetc_pf.c
index c48569444bf9..01725122a73f 100644
--- a/drivers/net/ethernet/freescale/enetc/enetc_pf.c
+++ b/drivers/net/ethernet/freescale/enetc/enetc_pf.c
@@ -104,7 +104,7 @@ static int enetc_mac_addr_hash_idx(const u8 *addr)
 		mask |= BIT_ULL(i * 6);
 
 	for (i = 0; i < 6; i++)
-		res |= (__sw_hweight64(fold & (mask << i)) & 0x1) << i;
+		res |= (hweight64(fold & (mask << i)) & 0x1) << i;
 
 	return res;
 }
@@ -284,7 +284,7 @@ static int enetc_vid_hash_idx(unsigned int vid)
 	int i;
 
 	for (i = 0; i < 6; i++)
-		res |= (__sw_hweight8(vid & (BIT(i) | BIT(i + 6))) & 0x1) << i;
+		res |= (hweight8(vid & (BIT(i) | BIT(i + 6))) & 0x1) << i;
 
 	return res;
 }
-- 
2.17.1

